src/tests/ftest/aggregation/continuous_write.py

```diff
@@ -0,0 +1,141 @@
+        """Repeatedly run IOR every second for 60 times or until aggregation is detected.
+
+        Args:
+            namespace (str): Namespace that defines block_size and transfer_size.
+            pool (TestPool): Pool to use with IOR.
+            container (TestContainer): Container to use with IOR.
+        """
+        ior_cmd = IorCommand(namespace=namespace)
+        ior_cmd.get_params(self)
+        ior_cmd.set_daos_params(self.server_group, pool, container.identifier)
+        testfile = os.path.join(os.sep, "test_file_1")
+        ior_cmd.test_file.update(testfile)
+        manager = get_job_manager(test=self, job=ior_cmd, subprocess=self.subprocess)
+        manager.assign_hosts(
+            self.hostlist_clients, self.workdir, self.hostfile_clients_slots)
+        ppn = self.params.get("ppn", namespace)
+        manager.assign_processes(ppn=ppn)
+
+        for count in range(60):
+            # In this test, use "##" in the log message because a thread is used and the logs are
+            # written in unpredictable manner. Using "##" makes it easier to find the debug logs
+            # without using search.
+            msg = f"## Run IOR. Count = {count}. Aggregation detected = {self.aggregation_detected}"
+            self.log.info(msg)
+            try:
+                manager.run()
+            except CommandFailure as error:
+                self.log.info(error)
+            if self.aggregation_detected:
+                self.log.info("## Aggregation detected. End IOR write loop.")
+                break
+            time.sleep(1)
+
+    def get_nvme_free(self, pool):
+        """Get NVMe free size from dmg pool query output.
+
+        Args:
+            pool (TestPool): Pool to query the NVMe free size.
+
+        Returns:
+            int: NVMe free size in Bytes. If not found in the pool query output, returns None.
+
+        """
+        query_out = pool.query()
+        tier_stats = query_out["response"]["tier_stats"]
+        for tier_stat in tier_stats:
+            if tier_stat["media_type"] == "nvme":
+                return tier_stat["free"]
+
+        return None
+
+    def test_continuous_write(self):
+        """Test that free space increase (aggregation) occurs while IOR is continuously running.
+
+        Aggregation should occur when the NVMe space becomes low. Test that aggregation occurs when
+        NVMe space becomes low due to many snapshots.
+
+        Testing aggregation properly isn't straightforward, especially using automation because the
+        algorithm is unclear and unpredictable. In this test, we focus on the occurrence of
+        aggregation while IOR write is continuously running at certain rate. I.e., We don't test the
+        amount of space reclaimed or different block sizes that fill up the NVMe at different rates.
+
+        Jira ID: DAOS-14204
+
+        1. Create a pool and a container.
+        2. Write a several MB file and repeatedly overwrite it with a thread.
```

Comment:
FYI that using ior flags `-w`, the default behavior is that IOR deletes the file after creating. So the file is created, deleted, recreated.
If you don't want the delete you can use flags `-w -k` (`-k` for "keep")

Reply:
Thanks. I tested both ways, and `-k` filled up the disk faster, but the time to trigger aggregation didn't change much, so I'll keep it this way.